### PR TITLE
Bug 1357668: update error message on invalid tags in ICTs

### PR DESCRIPTION
**Pull request by kargakis**
Also update the validation message for a missing tag for the istag name.

@openshift/api-review @mfojtik

### CODE REVIEW

**Comment by mfojtik** [comment]
@kargakis you said that the template was wrong :-)

**Comment by kargakis** [comment]
> @kargakis you said that the template was wrong :-)

It is wrong but we are defaulting unconditionally which is not correct:) Other than that the message needed fixing.

File: pkg/deploy/api/v1/defaults.go [hunk 1]
```diff
@@ -100,8 +100,10 @@ func SetDefaults_RollingDeploymentStrategyParams(obj *RollingDeploymentStrategyP
 func SetDefaults_DeploymentConfig(obj *DeploymentConfig) {
 	for _, t := range obj.Spec.Triggers {
 		if t.ImageChangeParams != nil {
-			t.ImageChangeParams.From.Kind = "ImageStreamTag"
```
Comment (deads2k) [inline pkg/deploy/api/v1/defaults.go]: So don't always default, but validation will fail if its not that value?  I can live with that and validation was already that tight as I recall.

However, didn't we have this running unconditionally in order to force the transformation of old data?

Comment (kargakis) [inline pkg/deploy/api/v1/defaults.go]: Oh, that's why we had it unconditionally... I am going to add a comment so I will remember next time:)

**Comment by mfojtik** [comment]
LGTM [merge]

**Pull request merged**